mautic/mautic
# Campaign Contact Stage persist

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
When a campaign action is triggered that changes anything on the contact, an error is produced because it tries to persist a Stage with null values. The Stage should not be persisted.
[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Try to trigger a campaign with a segment followed by change tags action and a send email action after 5 minutes. When the campaign is executed you will see an error like this < [Doctrine\DBAL\Exception\NotNullConstraintViolationException]                                                                                       
  An exception occurred while executing 'INSERT INTO stages (is_published, date_added, created_by, created_by_user, date_modified, modified_by, modi  
  fied_by_user, checked_out, checked_out_by, checked_out_by_user, name, description, weight, publish_up, publish_down, category_id) VALUES (?, ?, ?,  
   ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1, null, null, null, null, null, null, null, null, null, null, null, 0, null, null, null]:    
                                                                                                                                                      
  SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'name' cannot be null                                                                  
                                                                                                                                                      

                                                                                      
  [Doctrine\DBAL\Driver\PDOException]                                                 
  SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'name' cannot be null>


#### Steps to test this PR:
1. Apply this PR and run the campaign again, you should be able to run the campaign normally
